parlai/core/torch_agent.py

```diff
@@ -322,6 +322,10 @@ def load(self, path):
         Override this method for more specific loading.
         """
         states = torch.load(path, map_location=lambda cpu, _: cpu)
+        if 'model' in states:
+            self.model.load_state_dict(states['model'])
+        if 'optimizer' in states:
+            self.optimizer.load_state_dict(states['optimizer'])
```

Comment:
I think you need to also force some of the components to be `cuda` if we've enabled cuda already as seen [here](https://github.com/facebookresearch/ParlAI/blob/f64a694d1e10797a1b18c7a56633a6d90c4bf983/parlai/agents/seq2seq/seq2seq.py#L339).

Reply:
I think this may have been a hack for torch 0.3 having some missing functionality, let's try without it and see if it works :)

Reply:
Okay! Wasn't sure so just wanted to point it out.